pkg/cmd/roachtest/tests/cluster_to_cluster.go

```diff
@@ -1118,6 +1132,53 @@ func registerClusterReplicationResilience(r registry.Registry) {
 	}
 }
 
+func registerClusterReplicationDisconnect(r registry.Registry) {
+	sp := replicationSpec{
+		name:               "c2c/disconnect",
+		srcNodes:           3,
+		dstNodes:           3,
+		cpus:               4,
+		workload:           replicateKV{readPercent: 0, initRows: 1000000, maxBlockBytes: 1024},
+		timeout:            60 * time.Minute,
+		additionalDuration: 30 * time.Minute,
+		cutover:            2 * time.Minute,
+	}
+	c2cRegisterWrapper(r, sp, func(ctx context.Context, t test.Test, c cluster.Cluster) {
+		rd := makeReplicationDriver(t, c, sp)
+		rd.setupC2C(ctx, t, c)
+
+		shutdownSetupDone := make(chan struct{})
+
+		rd.replicationStartHook = func(ctx context.Context, rd *replicationDriver) {
+			defer close(shutdownSetupDone)
+		}
+		m := rd.newMonitor(ctx)
+		m.Go(func(ctx context.Context) error {
+			rd.main(ctx)
+			return nil
+		})
+
+		// Dont begin node disconnecion until c2c job is setup.
+		<-shutdownSetupDone
+		dstJobID := jobspb.JobID(getIngestionJobID(t, rd.setup.dst.sysSQL, rd.setup.dst.name))
+
+		// TODO(msbutler): disconnect nodes during a random phase
+		require.NoError(t, waitForTargetPhase(rd, dstJobID, phaseSteadyState))
+		sleepBeforeResiliencyEvent(rd)
+		ingestionProgress := getJobProgress(t, rd.setup.dst.sysSQL, dstJobID).GetStreamIngest()
+
+		srcDestConnections := getSrcDestNodePairs(rd, ingestionProgress)
+		randomNodePair := srcDestConnections[rand.Intn(len(srcDestConnections))]
+		rd.t.L().Printf("Disconnecting Src %d, Dest %d for the rest of the test", randomNodePair[0],
+			randomNodePair[1])
+
+		Disconnect(t, c, ctx, randomNodePair)
+		time.Sleep(time.Minute * 30)
```

Comment:
Should we add some expectations here? I could see either leaving it disconnected for some period of time and then reconnecting it and asserting that replication completes, or asserting that we eventually detect the disconnection and enter a paused state.

Reply:
The `m.wait()` call below gives the phys replication stream to completed. If it doesn't complete, the whole test should fail. I added some more commentary through the test to make this clear.